internal/bridge/managedpostgrescluster/managedpostgrescluster_controller.go

```diff
@@ -0,0 +1,552 @@
+						&managedPostgresClusterList.Items[index],
+					),
+				},
+			)
+		}
+
+		return reconcileRequests
+	})
+}
+
+//+kubebuilder:rbac:groups="postgres-operator.crunchydata.com",resources="managedpostgresclusters",verbs={get,patch,update}
+//+kubebuilder:rbac:groups="postgres-operator.crunchydata.com",resources="managedpostgresclusters/status",verbs={patch,update}
+//+kubebuilder:rbac:groups="postgres-operator.crunchydata.com",resources="managedpostgresclusters/finalizers",verbs={patch,update}
+//+kubebuilder:rbac:groups="",resources="secrets",verbs={get}
+
+// Reconcile does the work to move the current state of the world toward the
+// desired state described in a [v1beta1.ManagedPostgresCluster] identified by req.
+func (r *ManagedPostgresClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request) (result ctrl.Result, err error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Retrieve the managedpostgrescluster from the client cache, if it exists. A deferred
+	// function below will send any changes to its Status field.
+	//
+	// NOTE: No DeepCopy is necessary here because controller-runtime makes a
+	// copy before returning from its cache.
+	// - https://github.com/kubernetes-sigs/controller-runtime/issues/1235
+	managedpostgrescluster := &v1beta1.ManagedPostgresCluster{}
+	err = r.Get(ctx, req.NamespacedName, managedpostgrescluster)
+
+	if err == nil {
+		// Write any changes to the managedpostgrescluster status on the way out.
+		before := managedpostgrescluster.DeepCopy()
+		defer func() {
+			if !equality.Semantic.DeepEqual(before.Status, managedpostgrescluster.Status) {
+				status := r.Status().Patch(ctx, managedpostgrescluster, client.MergeFrom(before), r.Owner)
+
+				if err == nil && status != nil {
+					err = status
+				} else if status != nil {
+					log.Error(status, "Patching ManagedPostgresCluster status")
+				}
+			}
+		}()
+	} else {
+		// NotFound cannot be fixed by requeuing so ignore it. During background
+		// deletion, we receive delete events from managedpostgrescluster's dependents after
+		// managedpostgrescluster is deleted.
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// START SECRET HANDLING -- SPIN OFF INTO ITS OWN FUNC?
+
+	// Get and validate secret for req
+	key, team, err := r.GetAndCheckSecret(ctx, managedpostgrescluster)
+	if err != nil {
+		log.Error(err, "whoops, secret issue")
+
+		meta.SetStatusCondition(&managedpostgrescluster.Status.Conditions, metav1.Condition{
+			ObservedGeneration: managedpostgrescluster.GetGeneration(),
+			Type:               v1beta1.ConditionCreating,
+			Status:             metav1.ConditionFalse,
+			Reason:             "SecretInvalid",
+			Message: fmt.Sprintf(
+				"Cannot create with bad secret: %v", "TODO(managedpostgrescluster)"),
+		})
+
+		// Don't automatically requeue Secret issues
+		// We are watching for related secrets,
+		// so will requeue when a related secret is touched
+		return ctrl.Result{}, nil
+	}
+
```

Comment:
Should we check that the key is valid? Are there different levels of access in bridge? (Looks like API key doesn't have granular controls.)